js/packages/teams-ai/src/authentication/OAuthBotPrompt.ts

```diff
@@ -0,0 +1,143 @@
+ */
+export class OAuthBotPrompt extends OAuthPrompt {
+    private oauthSettings: OAuthSettings;
+
+    constructor(dialogId: string, settings: OAuthSettings) {
+        super(dialogId, settings);
+        this.oauthSettings = settings;
+    }
+
+    async beginDialog(dc: DialogContext): Promise<DialogTurnResult> {
+        // Ensure prompts have input hint set
+        const o: Partial<PromptOptions> = {
+            prompt: {
+                inputHint: InputHints.AcceptingInput
+            },
+            retryPrompt: {
+                inputHint: InputHints.AcceptingInput
+            }
+        };
+
+        // Initialize prompt state
+        const timeout = typeof this.oauthSettings.timeout === 'number' ? this.oauthSettings.timeout : 900000;
+        const state = dc.activeDialog!.state as OAuthPromptState;
+        state.state = {};
+        state.options = o;
+        state.expires = new Date().getTime() + timeout;
+
+        // Prompt user to login
+        await OAuthBotPrompt.sendOAuthCard(this.oauthSettings, dc.context, state.options.prompt);
+        return Dialog.EndOfTurn;
+    }
+
+    /**
+     * Sends an OAuth card.
+     * @param {OAuthPromptSettings} settings OAuth settings.
+     * @param {TurnContext} turnContext Turn context.
+     * @param {string | Partial<Activity>} prompt Message activity.
+     */
+    static override async sendOAuthCard(
+        settings: OAuthSettings,
+        turnContext: TurnContext,
+        prompt?: string | Partial<Activity>
+    ): Promise<void> {
+        // Initialize outgoing message
+        const msg: Partial<Activity> =
+            typeof prompt === 'object'
+                ? { ...prompt }
+                : MessageFactory.text(prompt ?? '', undefined, InputHints.AcceptingInput);
+
+        if (!Array.isArray(msg.attachments)) {
+            msg.attachments = [];
+        }
+
+        // Append appropriate card if missing
+        const msgHasOauthCardAttachment = msg.attachments.some(
+            (a) => a.contentType === CardFactory.contentTypes.oauthCard
+        );
+
+        if (!msgHasOauthCardAttachment) {
+            const cardActionType = ActionTypes.Signin;
+            const signInResource = await UserTokenAccess.getSignInResource(turnContext, settings);
+
+            let link = signInResource.signInLink;
+
+            if (
+                settings.showSignInLink === false ||
+                !this.signInLinkRequiredByChannel(turnContext.activity.channelId)
+            ) {
+                link = undefined;
+            }
+
+            let tokenExchangeResource;
+            if (settings.enableSso === true) {
+                // Send the tokene exchange resource only if enableSso is true.
+                tokenExchangeResource = signInResource.tokenExchangeResource;
+            }
+
+            // Append oauth card
+            const card = CardFactory.oauthCard(
```

Comment:
```suggestion
            const card = CardFactory.oAuthCard(
```

Reply:
This method is defined in bf sdk, i can't change it